docs/custom-pools.md

```diff
@@ -0,0 +1,154 @@
+# Custom pools
+
+Custom pools are pools that inherit from the `worker` pool. They use any MachineConfig targeted for the `worker` pool
+but add the ability to deploy changes only targeted at the custom pool.
+Since a custom pool inherits from the `worker` pool, any change to the `worker` pool is going to roll out to the custom pool as well (like an OS update during an upgrade).
```

Comment:
The worker config is the base for the infra node, and that's because infra is a superset of worker (look at the MachineConfig selector for the infra pool...). Also, a node is always a worker AND a custom pool and that's why the worker change reaches a custom pool.

Reply:
> How do we handle conflicts? say if have worker machineconfig saying to create a particular systemd file and infra pool asking the same file to be deleted, what should we do then?

So for this there's little we actually do right? This works with inheritance (and remember MCs are ordered lexicographically as well) so it's not really a conflict isn't it? it's something that you're doing cause you know the base pool has something you don't want. But maybe I'm missing something so happy to continue the discussion on this.

> Also, can we make worker pool generic enough so that it doesn't have info of the ignition, systemd units etc and let inherited pools have specialized info? This is to remove tight coupling from OS choice.

each pool _doesn't_ have knoledge of Ignition or anything platform specific. The pool is the grouping for MCs belonging to the pool itself (MCs are instead platform specific with ignition and whatnot).